### PR TITLE
setup status server for profiling

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: Yang Keao <keao.yang@yahoo.com>

## What have you changed? (mandatory)

Setup status server for profiling. You can set the address of status_server with `--status-server` argument.

Then you can get flamegraph or `pprof` protobuf through http request.

```
curl "http://STATUS_SERVER/debug/pprof/profile?frequency=50&seconds=10" # this will start cpu sampler at 10/sec frequency and continues 10 secs.
```

```
curl "http://STATUS_SERVER/debug/pprof/profile?frequency=50&seconds=10" --header  "Content-Type: application/protobuf" # this will return a pprof protobuf
```

## What are the type of the changes? (mandatory)

- New feature (change which adds functionality)

## How has this PR been tested? (mandatory)

No! I haven't test it. Is there anyone can help me test it?

## Does this PR affect TiDB Lightning? (mandatory)

No

### CODE REVIEW

File: src/bin/tikv-importer.rs [hunk 1]
```diff
@@ -88,16 +88,31 @@ fn main() {
...
+    tikv_config.log_file = config.log_file.clone();
+    tikv_config.log_rotation_timespan = config.log_rotation_timespan.clone();
+    initial_logger(&tikv_config);
+
+    if let Some(status_server_address) = matches.value_of("status-server") {
+        let mut status_server = tikv::server::status_server::StatusServer::new(1, tikv_config);
+
+        if let Err(e) = status_server.start(status_server_address.to_owned()) {
+            warn!("fail to setup status server: {:?}", e)
+        }
```
Comment (kennytm) [inline src/bin/tikv-importer.rs]: wouldn't the status_server be immediately dropped at the end?

Comment (YangKeao) [reply src/bin/tikv-importer.rs]: Oops! Yes. That's a mistake.

**Review by kennytm — COMMENTED** [review]
LGTM

**Review by overvenus — APPROVED** [review]

**Pull request merged**